src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,231 @@
+    IfFailRet(m_pProfilerManager->GetInstrumentationMethod(cslid, &pUnknown));
+    *ppUnknown = pUnknown.Detach();
+    return S_OK;
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::RemoveInstrumentationMethod(_In_ IInstrumentationMethod* pInstrumentationMethod)
+{
+    return m_pProfilerManager->RemoveInstrumentationMethod(pInstrumentationMethod);
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::AddInstrumentationMethod(
+    _In_ BSTR bstrModulePath,
+    _In_ BSTR bstrName,
+    _In_ BSTR bstrDescription,
+    _In_ BSTR bstrModule,
+    _In_ BSTR bstrClassGuid,
+    _In_ DWORD dwPriority,
+    _Out_ IInstrumentationMethod** ppInstrumentationMethod)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IInstrumentationMethod> pInstrumentationMethod;
+    IfFailRet(m_pProfilerManager->AddInstrumentationMethod(
+        bstrModulePath,
+        bstrName,
+        bstrDescription,
+        bstrModule,
+        bstrClassGuid,
+        dwPriority,
+        &pInstrumentationMethod));
+    *ppInstrumentationMethod = pInstrumentationMethod.Detach();
+    return S_OK;
+}
+
+// IProfilerManager2 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::DisableProfiling()
+{
+    return m_pProfilerManager->DisableProfiling();
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::ApplyMetadata(_In_ IModuleInfo* pMethodInfo)
+{
+    return m_pProfilerManager->ApplyMetadata(pMethodInfo);
+}
+
+// IProfilerManager3 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetApiVersion(_Out_ DWORD* pApiVer)
+{
+    HRESULT hr = S_OK;
+    DWORD apiVer;
+    IfFailRet(m_pProfilerManager->GetApiVersion(&apiVer));
+    *pApiVer = apiVer;
+    return S_OK;
+}
+
+// IProfilerManager4 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetGlobalLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IProfilerManagerLogging> pLogging;
+    IfFailRet(m_pProfilerManager->GetGlobalLoggingInstance(&pLogging));
+    *ppLogging = pLogging.Detach();
+    return S_OK;
+}
+
+
+// IProfilerManager5 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::IsInstrumentationMethodRegistered(_In_ REFGUID clsid, _Out_ BOOL* pfRegistered)
+{
+    HRESULT hr = S_OK;
+    BOOL fRegistered;
+    IfFailRet(m_pProfilerManager->IsInstrumentationMethodRegistered(clsid, &fRegistered));
+    *pfRegistered = fRegistered;
+    return S_OK;
+}
+
+// IProfilerManagerLogging Methods
+/*
+*/
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::LogMessage(_In_ const WCHAR* wszMessage)
+{
+    WCHAR wszPrefix[MAX_PATH];
+    wcscpy_s(wszPrefix, m_wszInstrumentationMethodGuid.c_str());
+    wcscat_s(wszPrefix, wszMessage);
+    return m_pProfilerManager->LogMessage(wszPrefix);
+}
+
+/*
+*/
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::LogError(_In_ const WCHAR* wszError)
+{
+    WCHAR wszPrefix[MAX_PATH];
```

Comment:
>wszPrefix[MAX_PATH] [](start = 10, length = 19)

Why MAX_PATH?

Reply:
It's the convention we use everywhere.

---
In reply to: [331698879](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r331698879) [](ancestors = 331698879)

Reply:
*almost everywhere. I'm fixing this to be consistent with the rest of the logging infrastructure (4096 characters)

---
In reply to: [331701234](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r331701234) [](ancestors = 331701234,331698879)